src/build.d

```diff
@@ -756,6 +849,132 @@ void processEnvironment()
+            "-Werror",
+            "-Wno-attributes",
+            "-Wno-char-subscripts",
+            "-Wno-deprecated",
+            "-Wno-empty-body",
+            "-Wno-format",
+            "-Wno-missing-braces",
+            "-Wno-missing-field-initializers",
+            "-Wno-overloaded-virtual",
+            "-Wno-parentheses",
+            "-Wno-reorder",
+            "-Wno-return-type",
+            "-Wno-sign-compare",
+            "-Wno-strict-aliasing",
+            "-Wno-switch",
+            "-Wno-type-limits",
+            "-Wno-unknown-pragmas",
+            "-Wno-unused-function",
+            "-Wno-unused-label",
+            "-Wno-unused-parameter",
+            "-Wno-unused-value",
+            "-Wno-unused-variable"
+        ];
+
+        if(kind == "g++") {
+            warnings ~= [
+                "-Wno-logical-op",
+                "-Wno-narrowing",
+                "-Wno-unused-but-set-variable",
+                "-Wno-uninitialized",
+                "-Wno-class-memaccess",
+                "-Wno-implicit-fallthrough"
+            ];
+        }
+    }
+    else
+    {
+        warnings = [
+            "-Wno-deprecated",
+            "-Wstrict-aliasing",
+            "-Werror"
+        ];
+
+        if(kind == "clang++")
+            warnings ~= "-Wno-logical-op-parentheses";
+    }
+
+    string hostDmdVernum;
+    if(env["HOST_DMD_KIND"] == "dmd") // Why limit this to dmd?
+    {
+        enum redirect = Redirect.stdin | Redirect.stdout | Redirect.stderrToStdout;
+        auto pipes = pipeProcess([env["HOST_DMD_RUN"], "-o-", "-c", "-"], redirect);
+
+        pipes.stdin.writeln("pragma(msg, cast(int)__VERSION__);");
+        pipes.stdin.flush();
+        pipes.stdin.close();
+        //  wait(pipes.pid);
+
+        hostDmdVernum = pipes.stdout.byLine.front.idup;
+    }
+    else {
+        hostDmdVernum = "2";
+    }
+
+    auto cxxFlags = warnings ~ [
+        "-fno-exceptions",
+        "-fno-rtti",
+        "-D__pascal=",
+        "-DMARS=1",
+        "-DTARGET_" ~ env["OS"].toUpper ~ "=1",
+        "-DDM_TARGET_CPU_" ~ env["TARGET_CPU"] ~ "=1",
+        env["MODEL_FLAG"],
+        env["PIC_FLAG"],
+        "-DDMD_VERSION=" ~ hostDmdVernum,
+
+        // No explicit if since kind will always be either g++ or clang++
+        kind == "g++" ? "-std=gnu++98" : "-xc++"
+    ];
+
+    const pgoDir = env.getDefault("PGO_DIR", srcDir ~ "/pgo");
+
+    const extraFlags = [
+        "ENABLE_DEBUG": ["-g", "-g3", "-DDEBUG=1", "-DUNITTEST"],
+        "ENABLE_RELEASE": ["-O2"],
+        "ENABLE_PROFILING": ["-pg", "-fprofile-arcs", "-ftest-coverage"],
+        "ENABLE_PGO_GENERATE": ["-fprofile-generate=" ~ pgoDir],
+        "ENABLE_PGO_USE": ["-fprofile-use=" ~ pgoDir, "-freorder-blocks-and-partition"],
+        "ENABLE_LTO": ["-flto"],
+        "ENABLE_UNITTEST": ["-unittest", "-cov"],
+        "ENABLE_COVERAGE": ["--coverage"],
+        "ENABLE_SANITIZERS": ["-fsanitize=" ~ env.getDefault("ENABLE_SANITIZERS", "")]
```

Comment:
Some of these aren't C++ options (i.e: `-unittest`, `-cov`).  Probably you also want to remove any environment flag variable that doesn't have a D equivalent as well.